src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,349 @@
+import {
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+  type ComponentProps,
+} from 'react'
+
+import { useTransitionState } from '../../hooks/useTransitionState.js'
+import ArrowLeftIcon from '../../icons/ArrowLeftIcon.js'
+import ArrowRightIcon from '../../icons/ArrowRightIcon.js'
+import ChevronLeftIcon from '../../icons/ChevronLeftIcon.js'
+import ChevronRightIcon from '../../icons/ChevronRightIcon.js'
+import { cn } from '../../utils/classnames.js'
+
+interface CarouselContext {
+  prevSlide: () => void
+  nextSlide: () => void
+  setSlide: (value: number) => void
+  totalSlides: number
+  currentSlide: number
+  autoAdvance: boolean
+  isPaused: boolean
+  setPaused: (value: boolean) => void
+  setDirection: (value: 'ltr' | 'rtl') => void
+  direction: 'ltr' | 'rtl'
+}
+
+const CarouselContext = createContext<CarouselContext | null>(null)
+
+function useCarouselState() {
+  const context = useContext(CarouselContext)
+
+  if (!context) {
+    throw new Error('useCarouselState must be used within the CarouselContext')
+  }
+  return context
+}
+
+function Carousel({
+  children,
+  count,
+  autoAdvance = false,
+  duration = 4000,
+}: {
+  children: React.ReactNode
+  count: number
+  autoAdvance?: boolean
+  duration?: number
+}) {
+  const [slide, setSlide] = useState(0)
+  const [isPaused, setPaused] = useState(false)
+  const [direction, setDirection] = useState<'rtl' | 'ltr'>('rtl')
+  function nextSlide() {
+    setSlide(current => {
+      setDirection('rtl')
+      let next = current
+
+      if (current === count - 1) {
+        next = 0
+      } else {
+        next = current + 1
+      }
+
+      return next
+    })
+  }
+
+  function prevSlide() {
+    setSlide(current => {
+      setDirection('ltr')
+
+      let prev = current
+      if (current === 0) {
+        prev = count - 1
+      } else {
+        prev = current - 1
+      }
+      return prev
+    })
+  }
+
+  const shouldAutoAdvance = duration > 0 || autoAdvance
```

Comment:
This seems superfluous, i think we can just get by with a single prop `duration` to trigger the auto advance.